caption/vtt.h

```diff
@@ -0,0 +1,87 @@
+/**********************************************************************************************/
+/* The MIT License                                                                            */
+/*                                                                                            */
+/* Copyright 2016-2017 Twitch Interactive, Inc. or its affiliates. All Rights Reserved.       */
+/*                                                                                            */
+/* Permission is hereby granted, free of charge, to any person obtaining a copy               */
+/* of this software and associated documentation files (the "Software"), to deal              */
+/* in the Software without restriction, including without limitation the rights               */
+/* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell                  */
+/* copies of the Software, and to permit persons to whom the Software is                      */
+/* furnished to do so, subject to the following conditions:                                   */
+/*                                                                                            */
+/* The above copyright notice and this permission notice shall be included in                 */
+/* all copies or substantial portions of the Software.                                        */
+/*                                                                                            */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR                 */
+/* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,                   */
+/* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE                */
+/* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER                     */
+/* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,              */
+/* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN                  */
+/* THE SOFTWARE.                                                                              */
+/**********************************************************************************************/
+#ifndef LIBCAPTION_VTT_H
+#define LIBCAPTION_VTT_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "caption.h"
+#include "eia608.h"
+
+// timestamp and duration are in seconds
```

Comment:
Perhaps we combine `vtt_t` and `srt_t` into a single `text_t` or `tt_t`. This will allow for more efficient conversions between str and vtt, and remove duplicate link list code

Reply:
One thing I'm unsure about is how best to represent things like REGION, STYLE and NOTE. It would be nice to have a data model that can contain all of these things.

Its valid to just ignore and discard NOTE blocks. But there needs to be a mechanism to parse and store REGION and STYLE.

Since REGION and STYLE blocks must occur before the first CUE, it seems reasonable to create a new structure to hold all the VTT information, and have linked-lists of CUE, REGION and STYLE blocks. This could then be used to print out a semantically-equivalent (but not text-equivalent) VTT.

Reply:
Make sense to me. Separation of `vtt_t`, and `text_t` (or `vtt_node_t` or whatever) the `vtt_t` object can also store head and tail, Thus simplifying the api. Can vtt be considered a superset of srt?

Reply:
I've created some new types, `vtt_block_t` and `vtt_t`, which contain pointers to the linked list of different blocks.

I've also updated the SRT examples to use VTT structs instead, but with wrapper functions which allows older code to still refer to things as SRT structs or `srt_cue_t` to make sure readers don't get confused.